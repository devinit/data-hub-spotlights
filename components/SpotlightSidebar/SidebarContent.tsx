import React, { FunctionComponent } from 'react';

interface SidebarContentProps {
  height?: string;
}

const SidebarContent: FunctionComponent<SidebarContentProps> = ({ children, height }) => {
  return (
    <div className="spotlight__content" style={{ height }}>
      {children}
      <style jsx>{`
        height: 100%;
      `}</style>
    </div>
  );
};

export { SidebarContent };
