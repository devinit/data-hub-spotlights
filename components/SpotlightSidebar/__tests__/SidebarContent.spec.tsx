/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SidebarContent } from '../SidebarContent';

describe('SidebarContent', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <SidebarContent/>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SidebarContent><div>My Child</div></SidebarContent>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
