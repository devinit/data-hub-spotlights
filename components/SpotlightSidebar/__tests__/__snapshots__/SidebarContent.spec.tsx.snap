// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarContent renders children correctly 1`] = `
<div
  className="spotlight__content"
  style={
    Object {
      "height": "400px",
    }
  }
>
  <div>
    My Child
  </div>
  <style
    jsx={true}
  >
    
        height: 100%;
      
  </style>
</div>
`;

exports[`SidebarContent renders correctly with the specified height style 1`] = `
<div
  className="spotlight__content"
  style={
    Object {
      "height": "300px",
    }
  }
>
  <style
    jsx={true}
  >
    
        height: 100%;
      
  </style>
</div>
`;
